src/primitives/types/types_typed.ts

```diff
@@ -61,6 +71,19 @@ export type Alignments = "left" | "middle" | "right" | number;
  */
 export type TransitionStates = "enter" | "visible" | "exiit" | number;
 
+/**
+ * A representation of a color. Can be:
+ *  - a string, which is parsed by THREE.Color
+ *  - a THREE.Color instance
+ *  - a number, which is interpreted as a hex value
+ *  - an array of numbers, which is interpreted as an RGB value
+ */
+type ColorDescription = string | Color | number | number[];
+
+type Vec2Like = number | number[] | Vector2;
```

Comment:
what's the meaning of the `number` type for these? Do we interpret that as the first coordinate and fill in zeros for the rest? I get it for color above, but not yet for the `Vec*Like` types.

Reply:
Basically, for vec2/3/4, supplying a single number behaves like supplying an array with one item. And if the array has fewer items than expected, default values are used for the remaining items.

Take for example `<cartesian />` nodes... One of their traits is `view3`. The `view3` trait defines several properties, [including `scale`](https://github.com/unconed/mathbox/blob/master/src/primitives/types/traits.ts#L40:
```js
// the Type for <cartesian>'s "scale" property, as defined in traits.js
scale: Types.vec3(1, 1, 1)
```
But imagine it was `scale: Types.vec3(a, b, c)`... Then:
```
cartesian.set("scale", x) // --> [x, b, c]
cartesian.set("scale", [x, y]) // --> [x, y, c]
cartesian.set("scale", [x, y, z]) // --> [x, y, z]
cartesian.set("scale", new THREE.Vector3(x)) // --> [x, 0, 0] ... the 0, 0 defaults are set by THREE
```

_I'm not trying to suggest that supplying a single number is actually a good idea... just documenting the current type validation system it seems intended._